security/permissions/src/main/java/io/quarkus/ts/openshift/security/basic/permissions/beanparam/NestedStructureBeanParam.java

```diff
@@ -0,0 +1,124 @@
+package io.quarkus.ts.openshift.security.basic.permissions.beanparam;
+
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.SecurityContext;
+import jakarta.ws.rs.core.UriInfo;
+
+public class NestedStructureBeanParam {
+    @QueryParam("userId")
+    private String userId;
+
+    @QueryParam("userName")
+    private String userName;
+
+    @QueryParam("resourceId")
+    private String resourceId;
+
+    @QueryParam("resourceType")
+    private String resourceType;
+
+    private User user;
+    private Resource resource;
+
+    @Context
+    private SecurityContext securityContext;
+
+    @Context
+    public UriInfo uriInfo;
+
+    private User getOrCreateUser() {
+        if (user == null) {
```

Comment:
Just a note, no need to modify this and other similar (I doing this sometime myself)

Sometimes it's more cleaner then nesting the if/else statement's to check for oposite and continue. Something like this

```java
if (user != null)  {
   return user;
}

user = new User();
if (userId != null) {
     user.id = userId;
}
if (userName != null) {
     user.name = userName;
}
```

In this case I don't think that it matter that much just it was something what I spot.

Reply:
Ok, "early return" is clearner, you're right, thanks